code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -6,113 +6,160 @@
 /datum/action/xeno_action/activable/transfer_plasma/drone
 	plasma_transfer_amount = PLASMA_TRANSFER_AMOUNT * 2
 
+/datum/action/xeno_action/create_jelly/slow
+	cooldown_timer = 45 SECONDS
+
 // ***************************************
-// *********** Acidic salve
+// *********** Essence Link
+// ***************************************
+/datum/action/xeno_action/activable/essence_link
+	name = "Essence Link"
+	action_icon_state = "healing_infusion"
+	mechanics_text = "Link to a xenomorph. This changes some of your abilities, and grants them and you both various bonuses."
+	cooldown_timer = 5 SECONDS
+	plasma_cost = 0
+	target_flags = XABB_MOB_TARGET
+	keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK
+	alternate_keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK_REMOVE
+	/// Used to determine whether there is an existing Essence Link or not. Also allows access to its vars.
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// The target of an existing link, if applicable.
+	var/mob/living/carbon/xenomorph/linked_target
+
+/datum/action/xeno_action/activable/essence_link/can_use_ability(mob/living/carbon/xenomorph/target, silent = FALSE, override_flags)
+	var/mob/living/carbon/xenomorph/X = owner
+	. = ..()
+	if(!.)
+		return FALSE
+	if(!isxeno(target) || target.hive.hivenumber != X.hive.hivenumber)
```

Comment:
1) do not cast without typechecking
2) there is a get_hivenumber proc that you need to use

Reply:
1 and 2 should be resolved.